kDriveCore/Utils/PHAsset/PHAsset+Exension.swift

```diff
@@ -23,29 +23,39 @@ import Photos
 import Sentry
 
 public extension PHAsset {
-    static func containsPhotosAvailableInHEIC(assetIdentifiers: [String]) -> Bool {
-        let assets = PHAsset.fetchAssets(withLocalIdentifiers: assetIdentifiers, options: nil)
-        var containsHEICPhotos = false
-        assets.enumerateObjects { asset, _, stop in
-            if let resource = asset.bestResource(), resource.uniformTypeIdentifier == UTI.heic.identifier {
-                containsHEICPhotos = true
-                stop.pointee = true
-            }
+    // MARK: - Hash
+
+    /// Get a hash of the base image of a PHAsset _without adjustments_
+    ///
+    /// Will return `nil` for any other resource type (like video)
+    var baseImageSHA256: String? {
+        guard let identifier = PHAssetIdentifier(self) else {
+            return nil
         }
-        return containsHEICPhotos
+
+        return identifier.baseImageSHA256
+    }
+
+    /// Hash of the best resource available. Editing a video or a picture will change this hash
+    var bestResourceSHA256: String? {
+        guard let identifier = PHAssetIdentifier(self) else {
+            return nil
+        }
+
+        return identifier.bestResourceSHA256
     }
 
     // MARK: - Filename
 
     /// Get a filename that can be used by kDrive, taking into consideration the edits that may exists on a PHAsset.
     func getFilename(fileExtension: String,
-                     creationDate: Date? = nil,
-                     modificationDate: Date? = nil,
-                     burstCount: Int? = nil,
```

Comment:
I removed the default parameters, there was a bug with a value not set somewhere it was possible to set the value.